File: Timofeev-PE-42/Bizzare-Underground-Java/Button.java
public class Button { // Не наследуется от GameObject в этом примере
    private boolean isPressed;
    private int cooldownButton;

    public Button() {
        this.isPressed = false;
        this.cooldownButton = 100;
        System.out.println("--- Button Initialized ---");
        System.out.println("  Cooldown: " + cooldownButton);
    }

    @Override
    protected void finalize() throws Throwable {
        System.out.println("--- Button Destroyed ---");
        super.finalize();
    }

    public boolean characterButtonContact(int characterPositionX, int characterPositionY) {
        System.out.println("Button.characterButtonContact() called. Character at (" + characterPositionX + ", " + characterPositionY + ") pressed button.");
        setIsPressed(true);
        return true;
    }

    public boolean boxButtonContact(int boxPositionX, int boxPositionY) {
        System.out.println("Button.boxButtonContact() called. Box at (" + boxPositionX + ", " + boxPositionY + ") pressed button.");
        setIsPressed(true);
        return true;
    }

    public boolean buttonChanges(boolean wallBlockIsHide, boolean spikeIsHide) {
        System.out.println("Button.buttonChanges() called. Applying changes.");
        System.out.println("  -> WallBlock hide status set to: " + wallBlockIsHide);
        System.out.println("  -> Spike hide status set to: " + spikeIsHide);
        // Логика изменения состояния других объектов
        return true;
    }

    // Getters
    public boolean getIsPressed() { return isPressed; }
    public int getCooldownButton() { return cooldownButton; }

    // Setters
    public void setIsPressed(boolean pressed) { this.isPressed = pressed; }
    public void setCooldownButton(int cooldown) { this.cooldownButton = cooldown; }
}

File: Timofeev-PE-42/Bizzare Underground C++/Project1/Button.cpp
#include "Button.h"
#include <iostream>

Button::Button() : isPressed(false), cooldownButton(100) {
    std::cout << "--- Button Initialized ---" << std::endl;
    std::cout << "  Cooldown: " << getCooldownButton() << std::endl;
}

Button::~Button() {
    std::cout << "--- Button Destroyed ---" << std::endl;
}

bool Button::characterButtonContact(int characterPositionX, int characterPositionY) {
    std::cout << "Button::characterButtonContact() called. Character at (" << characterPositionX << ", " << characterPositionY << ") pressed button." << std::endl;
    setIsPressed(true);
    return true;
}

bool Button::boxButtonContact(int boxPositionX, int boxPositionY) {
    std::cout << "Button::boxButtonContact() called. Box at (" << boxPositionX << ", " << boxPositionY << ") pressed button." << std::endl;
    setIsPressed(true);
    return true;
}

bool Button::buttonChanges(bool wallBlockIsHide, bool spikeIsHide) {
    std::cout << "Button::buttonChanges() called. Applying changes." << std::endl;
    std::cout << "WallBlock hide status set to: " << std::boolalpha << wallBlockIsHide << std::endl;
    std::cout << "Spike hide status set to: " << std::boolalpha << spikeIsHide << std::endl;
    return true;
}

bool Button::getIsPressed() const { return isPressed; }
int Button::getCooldownButton() const { return cooldownButton; }

void Button::setIsPressed(bool pressed) {
    isPressed = pressed;
    std::cout << "Button::setIsPressed() called. New state: isPressed = " << std::boolalpha << isPressed << std::endl;
}
void Button::setCooldownButton(int cooldown) {
    cooldownButton = cooldown;
    std::cout << "Button::setCooldownButton() called. New cooldown: " << cooldownButton << std::endl;
}